engine/src/contract_methods/connector/mod.rs

```diff
@@ -382,14 +384,15 @@ pub fn set_erc20_metadata<I: IO + Copy, E: Env, H: PromiseHandler>(
             io,
             env,
         );
-        let result = engine.set_erc20_metadata(args.erc20_address, args.erc20_metadata, handler)?;
+        let result = engine.set_erc20_metadata(&args.erc20_identifier, args.metadata, handler)?;
 
         Ok(result)
     })
 }
 
 pub fn get_erc20_metadata<I: IO + Copy, E: Env>(mut io: I, env: &E) -> Result<(), ContractError> {
-    let erc20_address = io.read_input_arr20().map(Address::from_array)?;
+    let erc20_identifier =
```

Comment:
Technically it is a breaking change, but I think it's okay because we haven't deployed this function yet.